src/core/thread/mle.cpp

```diff
@@ -636,13 +636,13 @@ otError Mle::BecomeDetached(void)
 {
     otError error = OT_ERROR_NONE;
 
-    VerifyOrExit(mRole != OT_DEVICE_ROLE_DISABLED, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(mRole != kRoleDisabled, error = OT_ERROR_INVALID_STATE);
```

Comment:
```suggestion
    VerifyOrExit(!IsDisabled(), error = OT_ERROR_INVALID_STATE);
```

Reply:
Thanks. Somehow I missed all the `mRole !=` checks in `Mle`. Will fix it... :)